src/bootstrap-table.js

```diff
@@ -2542,6 +2541,7 @@ class BootstrapTable {
 
     if (row) {
       this.options.data.splice(this.options.data.indexOf(row), 1)
```

Comment:
Should we alter the original data array?

Reply:
Updated the logic of the `removeByUniqueId` method to the same as the `remove` method.